File: src/SampleCRM.Web/Services/CountryCodesService.cs
using OpenRiaServices.DomainServices.Hosting;
using OpenRiaServices.DomainServices.Server;
using SampleCRM.Web.Attributes;
using SampleCRM.Web.Models;
using System.Data.Entity.Migrations;
using System.Linq;

namespace SampleCRM.Web
{
    [EnableClientAccess]
    public class CountryCodesService : SampleCRMService
    {
        [Query]
        public IQueryable<CountryCode> GetCountries() => 
            _context.CountryCodes.OrderBy(c => c.Name);

        public CountryCode GetCountryById(string countryCodeID) =>
            GetCountries().SingleOrDefault(x => x.CountryCodeID == countryCodeID);

        [Delete]
        [RestrictAccessReadonlyMode]
        public void DeleteCountry(CountryCode country) => _context.CountryCodes.Remove(country);

        [Insert]
        [RestrictAccessReadonlyMode]
        public void InsertCountry(CountryCode country) => _context.CountryCodes.AddOrUpdate(country);

        [Update]
        [RestrictAccessReadonlyMode]
        public void UpdateCountry(CountryCode country) => _context.CountryCodes.AddOrUpdate(country);
    }
}

File: src/SampleCRM.Web/Services/CountryCodesService.vb
Imports OpenRiaServices.DomainServices.Hosting
Imports OpenRiaServices.DomainServices.Server
Imports SampleCRM.Web.Attributes
Imports System.Data.Entity.Migrations
Imports System.Linq

Namespace SampleCRM.Web
    <EnableClientAccess>
    Public Class CountryCodesService
        Inherits SampleCRMService

        <Query>
        Public Function GetCountries() As IQueryable(Of CountryCodes)
            Return _context.CountryCodes
        End Function

        Public Function GetCountryById(ByVal countryCodeID As String) As CountryCodes
            Return _context.CountryCodes.SingleOrDefault(Function(x) x.CountryCodeID = countryCodeID)
        End Function

        <Delete>
        <RestrictAccessDeveloperMode>
        Public Sub DeleteCountry(ByVal country As CountryCodes)
            _context.CountryCodes.Remove(country)
        End Sub

        <Insert>
        <RestrictAccessDeveloperMode>
        Public Sub InsertCountry(ByVal country As CountryCodes)
            _context.CountryCodes.AddOrUpdate(country)
        End Sub

        <Update>
        <RestrictAccessDeveloperMode>
        Public Sub UpdateCountry(ByVal country As CountryCodes)
            _context.CountryCodes.AddOrUpdate(country)
        End Sub
    End Class
End Namespace
